esp-hal/src/rtc_cntl/sleep/esp32s3.rs

```diff
@@ -385,15 +359,39 @@ fn rtc_sleep_pu(val: bool) {
     bb.bbpd_ctrl()
         .modify(|_r, w| w.fft_force_pu().bit(val).dc_est_force_pu().bit(val));
 
-    register_modify_bits(
-        NRXPD_CTRL,
-        NRX_RX_ROT_FORCE_PU | NRX_VIT_FORCE_PU | NRX_DEMAP_FORCE_PU,
-        val,
-    );
+    if val {
+        nrx.nrxpd_ctrl().modify(|_, w| {
+            w.rx_rot_force_pu()
+                .set_bit()
+                .vit_force_pu()
+                .set_bit()
+                .demap_force_pu()
+                .set_bit()
+        });
+    } else {
+        nrx.nrxpd_ctrl().modify(|_, w| {
+            w.rx_rot_force_pu()
+                .clear_bit()
+                .vit_force_pu()
+                .clear_bit()
+                .demap_force_pu()
+                .clear_bit()
+        });
+    }
 
-    register_modify_bits(FE_GEN_CTRL, FE_IQ_EST_FORCE_PU, val);
+    if val {
+        fe.gen_ctrl().modify(|_, w| w.iq_est_force_pu().set_bit());
+    } else {
+        fe.gen_ctrl().modify(|_, w| w.iq_est_force_pu().clear_bit());
+    }
```

Comment:
```suggestion
    fe.gen_ctrl().modify(|_, w| w.iq_est_force_pu().bit(val));
```

Reply:
Oops, good catch, thanks! 😉

Reply:
There is also several places with similar case.